sort_by.rb

```diff
@@ -0,0 +1,47 @@
+# Defining  bubble_ Sort Defination
+def bubble_sort(array)
+	n = array.length
+	swapped = true	
+		while swapped do
+		swapped = false
+		(n - 1).times do |i|
+			if array[i] > array[i + 1]
+			array[i], array[i + 1] = array[i + 1], array[i]
+			swapped = true
+			end
+		end
+	end
+	puts "This is the sorted list #{array}"
+end
+
+# Inserting the list for sorting
+array = [5,9,7,3,1]
+puts "This is the list that needs to be sorted : #{array}"
+
+# Calling the function bubble_ Sort
+bubble_sort(array)
+
+# Defining  bubble_ Sort_by Defination
+def bubble_sort_by(array)
+	passes = array.length
+	while passes > 0
+		swapped = 1
+		while swapped < passes do
+			if yield(array[swapped-1],array[swapped]) < 0
+				array[swapped - 1], array[swapped] = array[swapped], array[swapped-1]
+			end
+			swapped += 1
+		end
+		passes -= 1
+	end
+	puts "This is the sorted list after sorting : #{array}"
```

Comment:
Let it just output only the array no strings

Reply:
@kobiyoyo Could you just explain it to me why DO i need to do the following , its not mentioned in the criteria that we cant have a string as out put.
I have made the changes but Still want to know what made you say that I need to just print array without strings ?